ofctrl/ofSwitch.go

```diff
@@ -304,3 +326,28 @@ func (self *OFSwitch) DumpFlowStats(cookieID, cookieMask uint64, flowMatch *Flow
 	}
 	return nil
 }
+
+func (self *OFSwitch) CheckStatus(timeout time.Duration) bool {
+	// Send echo request
+	self.changeStatus(false)
+	ch := make(chan bool)
+	go func() {
+		res := openflow13.NewEchoRequest()
+		self.Send(res)
+		for {
+			if self.IsReady() {
+				ch <- true
+				return
+			} else {
+				time.Sleep(50 * time.Millisecond)
+			}
```

Comment:
I don't really understand what's going on here, but it looks like a loop with a sleep in it that's waiting for a reply across a Unix domain socket. This is odd. I would normally use select() or poll() to wait for a reply, or a blocking read.

Reply:
I would add a channel to notify the controller has received echoReply message, and lever select() to wait for the notification or timeout event.